Ejercicio 1:

Contamos con los datos de consumo de agua de la CDMX obtenidos del portal de datos abiertos de la CDMX. Este set de datos supuestamente contiene "información bimestral de suministro de agua a nivel manzana, considerando la facturación por servicio de consumo medido y promedio"

Queremos generar el data profiling de estos datos.

In [60]:
import pandas as pd
import os

In [45]:
os.getcwd() #Da la ruta actual del directorio

'C:\\Users\\dor3l\\Documents\\Escuela\\ITAM\\CienciaDeDatos\\2019\\ICD'

In [46]:
print(os.listdir(os.getcwd())) #Genera listado de los archivos en la ruta

['.ipynb_checkpoints', 'consumo-agua.csv', 'consumo-agua.xls', 'ICD', 'Practica1_EDA.ipynb', 'Untitled-Copy1.ipynb']


In [47]:
df_consumo_agua=pd.read_csv('consumo-agua.csv',sep=';') #Carga CSV

In [80]:
#df_consumo_agua.sample(5) #Muestra aleatoria de 5 registros
df_consumo_agua.head() #Muestra los primeros 5 registros en la tabla de datos

Geo Point  \
0  19.4982924872,-99.1990438826   
1  19.4988910559,-99.2009080554   
2  19.4808107353,-99.2087681126   
3  19.4815775288,-99.2034616053   
4  19.4794098214,-99.2027692099   

                                           Geo Shape  consumo_total_mixto  \
0  {"type": "MultiPolygon", "coordinates": [[[[-9...               307.09   
1  {"type": "MultiPolygon", "coordinates": [[[[-9...                63.78   
2  {"type": "MultiPolygon", "coordinates": [[[[-9...               129.81   
3  {"type": "MultiPolygon", "coordinates": [[[[-9...               142.20   
4  {"type": "MultiPolygon", "coordinates": [[[[-9...                71.38   

   anio        nomgeo  consumo_prom_dom  consumo_total_dom      alcaldia  \
0  2019  Azcapotzalco         41.430000             455.73  AZCAPOTZALCO   
1  2019  Azcapotzalco         61.337142             429.36  AZCAPOTZALCO   
2  2019  Azcapotzalco         24.724412             840.63  AZCAPOTZALCO   
3  2019  Azcapotzalco         22.798462             592.76  AZCAPOTZALCO   
4  2019  Azcapotzalco         37.390334            1121.71  AZCAPOTZALCO   

        colonia  consumo_prom_mixto  consumo_total  consumo_prom  \
0  TIERRA NUEVA           51.181668        1029.50     54.184211   
1  TIERRA NUEVA           15.945000         530.88     40.836923   
2   SAN ANTONIO           64.905003         980.96     25.814737   
3   SAN ANTONIO          142.199997         766.23     27.365357   
4   SAN ANTONIO           35.690000        1203.12     36.458182   

   consumo_prom_no_dom  bimestre  consumo_total_no_dom    gid indice_des  
0               133.34         1                266.68  10206       BAJO  
1                18.87         1                 37.74  10203       BAJO  
2                 5.26         1                 10.52  10211      MEDIO  
3                31.27         1                 31.27  10217       BAJO  
4                10.03         1                 10.03  10220       BAJO

¿Cuántas variables tenemos?

¿Cuántas observaciones tenemos?


In [49]:
df_consumo_agua.shape #Indica el número de filas y columnas
#len(df_consumo_agua) #Observaciones
#df_consumo_agua.size #Total de elementos


(71102, 17)

17 variables y 71,102 observaciones

¿Cuántas observaciones únicas tenemos por variable?

In [51]:
df_consumo_agua.nunique()

Geo Point               22930
Geo Shape               22922
consumo_total_mixto     24339
anio                        1
nomgeo                     17
consumo_prom_dom        52060
consumo_total_dom       47051
alcaldia                   16
colonia                  1340
consumo_prom_mixto      31911
consumo_total           56015
consumo_prom            62214
consumo_prom_no_dom     37440
bimestre                    3
consumo_total_no_dom    27336
gid                     71102
indice_des                  4
dtype: int64

¿Cuántas variables numéricas tenemos?

¿Cuántas variables de fecha tenemos?

¿Cuántas variables categóricas tenemos?

¿Cuántas variables de texto tenemos?



In [57]:
df_consumo_agua.dtypes

Geo Point                object
Geo Shape                object
consumo_total_mixto     float64
anio                      int64
nomgeo                   object
consumo_prom_dom        float64
consumo_total_dom       float64
alcaldia                 object
colonia                  object
consumo_prom_mixto      float64
consumo_total           float64
consumo_prom            float64
consumo_prom_no_dom     float64
bimestre                  int64
consumo_total_no_dom    float64
gid                       int64
indice_des               object
dtype: object

In [61]:
df_consumo_agua.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 71102 entries, 0 to 71101
Data columns (total 17 columns):
Geo Point               71102 non-null object
Geo Shape               71078 non-null object
consumo_total_mixto     62775 non-null float64
anio                    71102 non-null int64
nomgeo                  71102 non-null object
consumo_prom_dom        66282 non-null float64
consumo_total_dom       66282 non-null float64
alcaldia                71102 non-null object
colonia                 71102 non-null object
consumo_prom_mixto      62775 non-null float64
consumo_total           71102 non-null float64
consumo_prom            71102 non-null float64
consumo_prom_no_dom     71102 non-null float64
bimestre                71102 non-null int64
consumo_total_no_dom    71102 non-null float64
gid                     71102 non-null int64
indice_des              71102 non-null object
dtypes: float64(8), int64(3), object(6)
memory usage: 9.2+ MB


Generea el profiling de cada variable

In [63]:
df_consumo_agua.describe() #Da el conteo de observaciones, media, desv. estándar mín, máx, quantiles excluyendo NaN

consumo_total_mixto     anio  consumo_prom_dom  consumo_total_dom  \
count         62775.000000  71102.0      66282.000000       66282.000000   
mean            174.359929   2019.0         29.132386        1186.263611   
std             312.663596      0.0         64.565925        2771.038307   
min               0.000000   2019.0          0.000000           0.000000   
25%               0.000000   2019.0         18.690547         161.635000   
50%              79.940000   2019.0         26.414248         604.185000   
75%             233.320000   2019.0         36.246563        1261.445000   
max           23404.440000   2019.0       7796.410000       95060.690000   

       consumo_prom_mixto  consumo_total  consumo_prom  consumo_prom_no_dom  \
count        62775.000000   71102.000000  71102.000000         71102.000000   
mean            50.636234    1695.847222    111.217399           126.760172   
std            130.408673    3555.697457   1069.949262          1095.817805   
min              0.000000       0.000000      0.000000             0.000000   
25%              0.000000     340.952500     23.010139             6.275417   
50%             33.451667     896.175000     31.693818            19.280000   
75%             61.216548    1808.902500     45.484917            54.186875   
max          11702.220000  119726.940000  89691.773438         89691.773438   

           bimestre  consumo_total_no_dom           gid  
count  71102.000000          71102.000000  71102.000000  
mean       2.006807            436.060309  35551.500000  
std        0.814393           2126.152162  20525.523757  
min        1.000000              0.000000      1.000000  
25%        1.000000             10.980000  17776.250000  
50%        2.000000             54.055000  35551.500000  
75%        3.000000            230.430000  53326.750000  
max        3.000000         119726.940000  71102.000000

¿Qué conocemos ahora de este set de datos por variable?

¿Cuántas alcadías tienes?

In [67]:
df_consumo_agua.alcaldia.nunique()

16

Son 16 alcaldías, sus valores únicos son:

In [69]:
df_consumo_agua.alcaldia.unique()

array(['AZCAPOTZALCO', 'VENUSTIANO CARRANZA', 'MILPA ALTA', 'TLALPAN',
       'ALVARO OBREGON', 'COYOACAN', 'BENITO JUAREZ', 'TLAHUAC',
       'MAGDALENA CONTRERAS', 'GUSTAVO A. MADERO', 'CUAJIMALPA',
       'IZTAPALAPA', 'XOCHIMILCO', 'IZTACALCO', 'CUAUHTEMOC',
       'MIGUEL HIDALGO'], dtype=object)

¿Cuántos nomgeo tienes?

In [70]:
df_consumo_agua.nomgeo.nunique()

17

Son 17 nomgeo, sus valores únicos son:

In [71]:
df_consumo_agua.nomgeo.unique()

array(['Azcapotzalco', 'Venustiano Carranza', 'Milpa Alta', 'Tlalpan',
       'Álvaro Obregón', 'Coyoacán', 'Benito Juárez', 'Tláhuac',
       'La Magdalena Contreras', 'Gustavo A. Madero',
       'Cuajimalpa de Morelos', 'Iztapalapa', 'Xochimilco', 'Iztacalco',
       'Cuauhtémoc', 'Talpan', 'Miguel Hidalgo'], dtype=object)

¿Identificas algún error?
Sí, hay un error de captura en la alcaldía de Tlalpan con Talpan


 To Do, 26 septiembre 2019

Transformar las variables a formato estándar: minúsculas, sin espacios en blanco, sin signos de puntuación.


Geo Point  \
0      19.4982924872,-99.1990438826   
1      19.4988910559,-99.2009080554   
2      19.4808107353,-99.2087681126   
3      19.4815775288,-99.2034616053   
4      19.4794098214,-99.2027692099   
5        19.4975265487,-99.21195176   
6      19.4966164619,-99.1930681289   
7       19.4900746324,-99.196630624   
8      19.4840235824,-99.2103915833   
9       19.498490991,-99.2009884028   
10     19.4885632293,-99.2031811727   
11     19.4838426741,-99.2120563592   
12     19.4885632293,-99.2031811727   
13     19.4938030532,-99.2140930535   
14     19.4955312206,-99.1996191463   
15     19.4311310364,-99.1210559573   
16     19.4972213126,-99.2115731442   
17     19.4967520855,-99.2105402569   
18     19.4968695952,-99.2097730685   
19     19.4956006378,-99.2097079295   
20      19.480339316,-99.2192385683   
21      19.4958193012,-99.210833036   
22     19.4964979586,-99.2127640154   
23     19.5046778297,-99.2091968959   
24     19.5025781461,-99.2076315381   
25     19.4819085048,-99.2176866064   
26     19.5033712292,-99.2116643343   
27     19.4805521248,-99.2131005626   
28     19.4797661242,-99.2133421378   
29     19.4790181216,-99.2142809819   
...                             ...   
71072  19.4583640163,-99.1070404251   
71073  19.4567254015,-99.1053996873   
71074  19.4564245304,-99.1040551197   
71075  19.4562262397,-99.1031750623   
71076  19.4561304349,-99.1027461868   
71077  19.4543539316,-99.1019622946   
71078   19.454861607,-99.1041911333   
71079  19.4511224182,-99.1037815386   
71080  19.4554542169,-99.1068327497   
71081  19.4572208753,-99.1089372428   
71082  19.4509232756,-99.1032458849   
71083  19.4577164392,-99.1102587235   
71084  19.4587749696,-99.1146564977   
71085  19.4587749696,-99.1146564977   
71086  19.4587749696,-99.1146564977   
71087  19.4589967703,-99.1156490038   
71088  19.4593008164,-99.1170178529   
71089  19.4566356467,-99.1128713893   
71090  19.4563895361,-99.1118870498   
71091  19.4563034696,-99.1113661735   
71092  19.4556020491,-99.1096105214   
71093  19.4537760119,-99.1049216798   
71094  19.4556020491,-99.1096105214   
71095  19.4556020491,-99.1096105214   
71096  19.4545520265,-99.1075168129   
71097   19.4516942696,-99.102942796   
71098  19.4523705698,-99.1047431661   
71099  19.4526041318,-99.1053626123   
71100  19.4530852681,-99.1066353366   
71101  19.4535621308,-99.1078964522   

                                               Geo Shape consumo_total_mixto  \
0      {"type": "multipolygon", "coordinates": [[[[-9...              307.09   
1      {"type": "multipolygon", "coordinates": [[[[-9...               63.78   
2      {"type": "multipolygon", "coordinates": [[[[-9...              129.81   
3      {"type": "multipolygon", "coordinates": [[[[-9...               142.2   
4      {"type": "multipolygon", "coordinates": [[[[-9...               71.38   
5      {"type": "multipolygon", "coordinates": [[[[-9...                 nan   
6      {"type": "multipolygon", "coordinates": [[[[-9...               108.0   
7      {"type": "multipolygon", "coordinates": [[[[-9...               152.8   
8      {"type": "multipolygon", "coordinates": [[[[-9...              140.45   
9      {"type": "multipolygon", "coordinates": [[[[-9...              269.05   
10     {"type": "multipolygon", "coordinates": [[[[-9...              238.37   
11     {"type": "multipolygon", "coordinates": [[[[-9...              765.43   
12     {"type": "multipolygon", "coordinates": [[[[-9...                 nan   
13     {"type": "multipolygon", "coordinates": [[[[-9...              177.02   
14     {"type": "multipolygon", "coordinates": [[[[-9...                 nan   
15     {"type": "multipolygon", "coordinates": [[[[-9...                 nan   
16     {"type": "multipolygon", "coordinates": [[[[-9...               22.42   
17     {"type": "multipolygon", "coordinates": [[[[-9...                 nan   
18     {"type": "multipolygon", "coordinates": [[[[-9...               42.48   
19     

In [82]:
df_consumo_agua.apply(lambda x: x.astype(str).str.lower()) #convierte a minúsculas las observaciones de tipo string

Geo Point  \
0      19.4982924872,-99.1990438826   
1      19.4988910559,-99.2009080554   
2      19.4808107353,-99.2087681126   
3      19.4815775288,-99.2034616053   
4      19.4794098214,-99.2027692099   
5        19.4975265487,-99.21195176   
6      19.4966164619,-99.1930681289   
7       19.4900746324,-99.196630624   
8      19.4840235824,-99.2103915833   
9       19.498490991,-99.2009884028   
10     19.4885632293,-99.2031811727   
11     19.4838426741,-99.2120563592   
12     19.4885632293,-99.2031811727   
13     19.4938030532,-99.2140930535   
14     19.4955312206,-99.1996191463   
15     19.4311310364,-99.1210559573   
16     19.4972213126,-99.2115731442   
17     19.4967520855,-99.2105402569   
18     19.4968695952,-99.2097730685   
19     19.4956006378,-99.2097079295   
20      19.480339316,-99.2192385683   
21      19.4958193012,-99.210833036   
22     19.4964979586,-99.2127640154   
23     19.5046778297,-99.2091968959   
24     19.5025781461,-99.2076315381   
25     19.4819085048,-99.2176866064   
26     19.5033712292,-99.2116643343   
27     19.4805521248,-99.2131005626   
28     19.4797661242,-99.2133421378   
29     19.4790181216,-99.2142809819   
...                             ...   
71072  19.4583640163,-99.1070404251   
71073  19.4567254015,-99.1053996873   
71074  19.4564245304,-99.1040551197   
71075  19.4562262397,-99.1031750623   
71076  19.4561304349,-99.1027461868   
71077  19.4543539316,-99.1019622946   
71078   19.454861607,-99.1041911333   
71079  19.4511224182,-99.1037815386   
71080  19.4554542169,-99.1068327497   
71081  19.4572208753,-99.1089372428   
71082  19.4509232756,-99.1032458849   
71083  19.4577164392,-99.1102587235   
71084  19.4587749696,-99.1146564977   
71085  19.4587749696,-99.1146564977   
71086  19.4587749696,-99.1146564977   
71087  19.4589967703,-99.1156490038   
71088  19.4593008164,-99.1170178529   
71089  19.4566356467,-99.1128713893   
71090  19.4563895361,-99.1118870498   
71091  19.4563034696,-99.1113661735   
71092  19.4556020491,-99.1096105214   
71093  19.4537760119,-99.1049216798   
71094  19.4556020491,-99.1096105214   
71095  19.4556020491,-99.1096105214   
71096  19.4545520265,-99.1075168129   
71097   19.4516942696,-99.102942796   
71098  19.4523705698,-99.1047431661   
71099  19.4526041318,-99.1053626123   
71100  19.4530852681,-99.1066353366   
71101  19.4535621308,-99.1078964522   

                                               Geo Shape consumo_total_mixto  \
0      {"type": "multipolygon", "coordinates": [[[[-9...              307.09   
1      {"type": "multipolygon", "coordinates": [[[[-9...               63.78   
2      {"type": "multipolygon", "coordinates": [[[[-9...              129.81   
3      {"type": "multipolygon", "coordinates": [[[[-9...               142.2   
4      {"type": "multipolygon", "coordinates": [[[[-9...               71.38   
5      {"type": "multipolygon", "coordinates": [[[[-9...                 nan   
6      {"type": "multipolygon", "coordinates": [[[[-9...               108.0   
7      {"type": "multipolygon", "coordinates": [[[[-9...               152.8   
8      {"type": "multipolygon", "coordinates": [[[[-9...              140.45   
9      {"type": "multipolygon", "coordinates": [[[[-9...              269.05   
10     {"type": "multipolygon", "coordinates": [[[[-9...              238.37   
11     {"type": "multipolygon", "coordinates": [[[[-9...              765.43   
12     {"type": "multipolygon", "coordinates": [[[[-9...                 nan   
13     {"type": "multipolygon", "coordinates": [[[[-9...              177.02   
14     {"type": "multipolygon", "coordinates": [[[[-9...                 nan   
15     {"type": "multipolygon", "coordinates": [[[[-9...                 nan   
16     {"type": "multipolygon", "coordinates": [[[[-9...               22.42   
17     {"type": "multipolygon", "coordinates": [[[[-9...                 nan   
18     {"type": "multipolygon", "coordinates": [[[[-9...               42.48   
19     

Agregar la variable latitud y longitud.


Pasar la variable latitud y longitud a numérica -si no la tomó como numérica-.


Eliminar la columna geo_point -una vez que creaste la variable latitud y longitud.

Eliminar la columna geo_shape.


Cambiar a minúsculas las columnas alcaldía, colonia e indice_des.


In [ ]:
df_consumo_agua.apply(lambda x: x.astype(str).str.lower()) #convierte a minúsculas las observaciones de tipo stri

Volver a correr el proceso de identificación de variables numéricas, categóricas, texto y fechas.


Genera el data profiling por variable.